nwc_backend/db.py

```diff
@@ -142,6 +143,20 @@ def provide_token(cparams: dict[str, Any], **_kwargs: Any) -> None:
             token_cache.extend((monotonic(), token))
         cparams["password"] = token_cache[1]
 
+        # SQLAlchemy converts the URL to connect() arguments, but asyncpg
+        # only accepts sslmode et al. in a URL, not as arguments. So we
+        # need to construct an SSL context instead.
+        if "ssl" not in cparams and engine.url.drivername == "postgresql+asyncpg":
```

Comment:
As far as I can tell, this is the right drivername here, but let me know if you'd prefer some other method of getting this @mgorven